src/Inline/Parser/CloseBracketParser.php

```diff
@@ -89,26 +82,29 @@ public function parse(ContextInterface $context, InlineParserContext $inlineCont
             return false;
         }
 
+        $node = $this->createInline($link['url'], $link['title'], $isImage);
+        $openerNode = $opener->getInlineNode();
+        $openerNode->insertBefore($node);
+        while (($inline = $openerNode->getNext()) !== null) {
+            $node->appendChild($inline);
+        }
+        $openerNode->detach();
+
```

Comment:
Would you mind describing these changes for me?  It might be quicker for you to explain them than for me to reason them out myself :)